src/java/org/apache/cassandra/io/sstable/format/SSTableWriter.java

```diff
@@ -385,4 +396,43 @@ public abstract SSTableWriter open(Descriptor descriptor,
                                            Collection<SSTableFlushObserver> observers,
                                            LifecycleNewTracker lifecycleNewTracker);
     }
+
+    public static void guardCollectionSize(TableMetadata metadata, DecoratedKey partitionKey, Unfiltered unfiltered)
+    {
+        if (!unfiltered.isRow() || SchemaConstants.isSystemKeyspace(metadata.keyspace))
+            return;
+
+        if (!Guardrails.collectionSize.enabled(null) && !Guardrails.itemsPerCollection.enabled(null))
```

Comment:
I would put this check first so we spare the previous `if` on nodes where Guardrails is disabled. Otherwise now we are incurring a cost needlessly?

Reply:
Good catch, `Unfiltered#isRow` should be as cheap as checking the guardrails, but `SchemaConstants.isSystemKeyspace` contains multiple calls to `Set#contains` that are more costly. Just changed the ordering of those `if`.